SwanProjects/src/ProjectDialog.tsx

```diff
@@ -0,0 +1,218 @@
+              'SWAN_projects/' + options.name
+            ).catch((): void => {
+             // No message here, it is not needed,
+              //I am checking if the directory doesn't exist in order
+              //to make valid the creation of the project folder.
+            });
+            if (content === undefined) {
+              valid = true;
+            } else {
+              await showErrorMessage(
+                'Invalid project name',
+                'Project already exists.'
+              );
+              valid = false;
+            }
+          } else {
+            //this is a special case for editing because I need to check that the new name of the project doesn't exists.
+            if (old_options.name !== options.name) {
+              const content = await contentRequest(
+                'SWAN_projects/' + options.name
+              ).catch(() => {
+                // No message here, it is not needed,
+                //I am checking if the directory doesn't exist in order
+                //to make valid the edition of the name of the project folder.
+              });
+              if (content === undefined) {
+                valid = true; //folder doesn't exists, then I can to raname the project.
+              } else {
+                await showErrorMessage(
+                  'Invalid project name',
+                  'Project already exists.'
+                );
+                valid = false;
+                continue;
+              }
+            }
+
+            if (options.corrupted) {
+              valid = true;
+              break;
+            }
+
+            //verifying that options where changed, othewise I will not send the request
+            if (JSON.stringify(old_options) !== JSON.stringify(options)) {
+              valid = true;
+            } else {
+              valid = false;
+            }
+          }
+        }
+
+        if (options.name?.trim() === '') {
+          await showErrorMessage(
+            'Invalid project name',
+            'Select a valid (non-empty) project name.'
+          );
+          valid = false;
+        }
+      } else {
+        valid = true;
+      }
+    } while (!valid);
+    if (dialogResult.changesSaved) {
+      startSpinner();
+      if (create) {
+        await createProjectRequest(options)
+          .then((res: ISWANReqResponse) => {
+            if (res.status) {
+              commands.execute('filebrowser:go-to-path', {
+                path: res.project_dir,
+                showBrowser: false
+              });
+            } else {
+              stopSpinner();
+              showErrorMessage('Error creating project', res.msg);
+            }
+            return res;
+          })
+          .catch((msg: any): void => {
+            stopSpinner();
+            showErrorMessage('Error creating project', msg);
+          });
+      } else {
+        await commands
+          .execute('filebrowser:go-to-path', {
```

Comment:
There is a special case when the project is edited and the name is changed.
As soon as the project is renamed, the file browser produces an error explaining that the folder was removed.
To avoid this, only when the name of the project is changed, I am going one folder back to do the rename and I am going to the new folder project when the promise is finished, to leave the user in the new project folder.